packages/lesswrong/server/editor/make_editable_callbacks.js

```diff
@@ -145,7 +145,7 @@ export function addEditableCallbacks({collection, options = {}}) {
       newModifier.$set = {...modifier.$set, ...newFields}
       if (modifier.$unset) {delete modifier.$unset.htmlBody}
     } else if (modifier.$set && modifier.$set.htmlBody) {
-      newFields = convertFromHTML(modifier.$set.htmlBody, !author.isAdmin, fieldName);
+      newFields = convertFromHTML(modifier.$set.htmlBody, !(author && author.isAdmin), fieldName);
```

Comment:
It looks like the callsite of this is `vulcan-lib/server/mutators.js:222` where we have a line that says

```
  data = modifierToData(await runCallbacks(`${collectionName.toLowerCase()}.edit.sync`, dataToModifier(data), document, currentUser, newDocument));
```

I don't know how `currentUser` could be null in that context; the editable entity types that this callback could be used for are comments, local groups, messages, and posts, none of which allow anonymous editing, and this is well past the validation callback. So I think there's a deeper problem here, and this bit of defensive programming is more likely to open a security vulnerability than it is to fix it.

Reply:
Hypothesis: this happens if you write a post in the editor, keep the tab open, get logged out by the server forgetting your login token, then click Submit. This should fail with a validation error, but is now instead failing with this null-dereference error. This is nevertheless better than not-failing, which would cause an authorless post to be created.